src/threading.ts

```diff
@@ -117,12 +118,24 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(exc: IPy_Object): void {
+        // Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+        this.exc = exc;
+        var f: IPy_FrameObj = this.getTopOfStack();
+        if (!f.tryCatchException(this)) {
+            // Find in previous frames
+            for (var i = this.stack.length - 2; i >= 0; i--) {
```

Comment:
I was leaving them on. If I pop them off, I have a single failing test. The failure occurs inside of ```DUP_TOP``` opcode when it tries to push an undefined value on the frame stack. (because the frame stack was empty when ```DUP_TOP``` was encountered)
https://github.com/Mdrbhatti/Ninia/blob/master/src%2Fframeobject.ts#L94

Removing the assertion results in a passing test.